lib/watcher-middleware.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var url = require('url');
+var path = require('path');
+
+var setResponseHeaders = require('./utils/response-header');
+var errorHandler = require('./utils/error-handler');
+
+module.exports = function watcherMiddleware(watcher, options) {
+  options = options || {};
+
+  if (!options.hasOwnProperty('autoIndex')) {
+    // set autoIndex to be true if not provided
+    options.autoIndex = true;
+  }
+
+  return function(request, response, next) {
+    watcher.then(function(hash) {
+      var outputPath = path.normalize(hash.directory);
+
+      // set the x-broccoli header containing per request info used by the broccoli-middleware
+      var urlToBeServed = request.url;
+      var urlObj = url.parse(urlToBeServed);
+      var filename = path.join(outputPath, decodeURIComponent(urlObj.pathname));
+      var broccoliInfo = {
+        'url': urlToBeServed,
+        'outputPath': outputPath,
+        'filename': filename,
+      };
+      request.headers['x-broccoli'] = broccoliInfo;
+
+      var updatedFileName = setResponseHeaders(request, response, next, {
```

Comment:
A couple of points here:

1. I'm surprised to see so much of the response header set here.  eg `setResponseHeaders` sets things like `content-length` which seems to be a concern of the `serveAssetMiddleware` not the `watcherMiddleware`.  In [this comment](https://github.com/ember-cli/rfcs/pull/80#pullrequestreview-9709454) the only headers set up in the watcher middleware were the `x-broccoli` request header for tracking the output path and the `Cache-Control` response header as that's a pretty reasonable header to set as a default irrespective of how the request is actually fulfilled.

Why are file-specific headers set in this middleware rather than the asset serving middleware?

2. It seems weird for `setResponseHeaders` to not be a middleware but to still accept `next`.  I'm not suggesting that it should be a middleware: I'm worried that it's currently hiding some of the control flow.  Is there some way we can split up this function?